### PR TITLE
Apply NumPy docstyle to remaining files in "signac" root.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR applies NumPy docstyle to remaining files in the `signac` directory (i.e. the level of `__init__.py`).

## Motivation and Context
Progress on #315. After this, remaining PRs should remove `core`, `common`, and `contrib` from the "skipped" directories in `match-dir` in `setup.cfg`.

## Types of Changes

- [x] Documentation update
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added all related issue and pull request numbers for future reference (if applicable). See example below.

### CODE REVIEW

File: signac/sync.py [hunk 1]
```diff
@@ -254,68 +259,58 @@ def sync_jobs(src, dst, strategy=None, exclude=None, doc_sync=None, recursive=Fa
...
+    dst : :class:`~signac.contrib.job.Job`
+        The dst job, data will be copied to this job's workspace.
+    strategy :
+        A synchronization strategy for file conflicts. If no strategy is
+        provided, a :class:`.errors.SyncConflict` exception will be raised upon
+        conflict. (Default value = None)
+    exclude : str
+        A filename exclusion pattern. All files matching this pattern will be
+        excluded from the synchronization process. (Default value = None)
+    doc_sync :
```
Comment (cbkerr) [inline signac/sync.py:290]: Do `doc_sync` and `strategy` need a type?

File: signac/sync.py [hunk 2]
```diff
@@ -393,67 +388,68 @@ def sync_projects(source, destination, strategy=None, exclude=None, doc_sync=Non
...
-    :param check_schema:
+    strategy :
+        A file synchronization strategy. (Default value = None)
+    exclude :
+        Files with names matching the given pattern will be excluded from the
+        synchronization. (Default value = None)
+    doc_sync :
+        The function applied for synchronizing documents. (Default value =
+        None)
+    selection :
```
Comment (cbkerr) [inline signac/sync.py:405]: Do the optional arguments need types listed for their non-default values?

File: signac/syncutil.py [hunk 3]
```diff
@@ -15,14 +20,30 @@
...
-    "Implementation adapted from https://docs.python.org/3/library/shutil.html#copytree-example'."
+    """Recursively copy a directory tree from src to dst, using a custom copy function.
+
+    Implementation adapted from https://docs.python.org/3/library/shutil.html#copytree-example.
+
+    Parameters
+    ----------
+    src :
+        Source directory tree.
+    dst :
```
Comment (cbkerr) [inline signac/syncutil.py:39]: if I'm reading the code correctly:
```suggestion
    dst : str
```

File: signac/syncutil.py [hunk 4]
```diff
@@ -15,14 +20,30 @@
...
 
 def copytree(src, dst, copy_function=shutil.copy2, symlinks=False):
-    "Implementation adapted from https://docs.python.org/3/library/shutil.html#copytree-example'."
+    """Recursively copy a directory tree from src to dst, using a custom copy function.
+
+    Implementation adapted from https://docs.python.org/3/library/shutil.html#copytree-example.
+
+    Parameters
+    ----------
+    src :
```
Comment (cbkerr) [inline signac/syncutil.py:37]:
```suggestion
    src : str
```

File: signac/sync.py (hunk 1 above)
Comment (bdice) [reply signac/sync.py:290]: I have no idea what to put for the type of `doc_sync`. It supports `None`, `False`, `'copy'`, and callables. Users are supposed to provide one of the properties/functors from the `DocSync` class, but I'm not sure how to express that as a type. @csadorf Do you have thoughts? Maybe this?
```suggestion
    doc_sync : attribute or callable provided by :py:class:`~signac.sync.DocSync`
```

Comment (bdice) [reply signac/sync.py:290]: I went ahead and committed a slightly edited version of that suggestion above.

**Comment by bdice** [comment]
@cbkerr Excellent suggestions, thanks for catching the things I missed. I agree we should adopt the use of the "optional" keyword, but I realize that it is missing in many many places. I will create a separate issue for that so that we remember to apply it everywhere once our docstrings have all been converted.

File: signac/sync.py (hunk 1 above)
Comment (csadorf) [reply signac/sync.py]: I think that's fine, but maybe you can add an example on how to use this function as part of the general description? I think that would make it much easier for users to understand how to use this.

**Pull request merged**